coresimd/aarch64/crypto.rs

```diff
@@ -0,0 +1,112 @@
+use coresimd::simd::*;
+
+#[allow(improper_ctypes)]
+extern "C" {
+    #[link_name = "llvm.aarch64.crypto.sha1h"]
+    fn vsha1h_u32_(hash_e: u32) -> u32;
+    #[link_name = "llvm.aarch64.crypto.sha1su0"]
+    fn vsha1su0q_u32_(w0_3: u32x4, w4_7: u32x4, w8_11: u32x4) -> u32x4;
+    #[link_name = "llvm.aarch64.crypto.sha1su1"]
+    fn vsha1su1q_u32_(tw0_3: u32x4, w12_15: u32x4) -> u32x4;
+    #[link_name = "llvm.aarch64.crypto.sha1c"]
+    fn vsha1cq_u32_(hash_abcd: u32x4, hash_e: u32, wk: u32x4) -> u32x4;
+    #[link_name = "llvm.aarch64.crypto.sha1p"]
+    fn vsha1pq_u32_(hash_abcd: u32x4, hash_e: u32, wk: u32x4) -> u32x4;
+    #[link_name = "llvm.aarch64.crypto.sha1m"]
+    fn vsha1mq_u32_(hash_abcd: u32x4, hash_e: u32, wk: u32x4) -> u32x4;
+
+    #[link_name = "llvm.aarch64.crypto.sha256h"]
+    fn vsha256hq_u32_(hash_abcd: u32x4, hash_efgh: u32x4, wk: u32x4) -> u32x4;
+    #[link_name = "llvm.aarch64.crypto.sha256h2"]
+    fn vsha256h2q_u32_(hash_efgh: u32x4, hash_abcd: u32x4, wk: u32x4) -> u32x4;
+    #[link_name = "llvm.aarch64.crypto.sha256su0"]
+    fn vsha256su0q_u32_(w0_3: u32x4, w4_7: u32x4) -> u32x4;
+    #[link_name = "llvm.aarch64.crypto.sha256su1"]
+    fn vsha256su1q_u32_(tw0_3: u32x4, w8_11: u32x4, w12_15: u32x4) -> u32x4;
+}
+
+#[cfg(test)]
+use stdsimd_test::assert_instr;
+
+/// SHA1 hash update accelerator, choose.
+#[inline]
+#[target_feature(enable = "crypto")]
+pub unsafe fn vsha1cq_u32(hash_abcd: u32x4, hash_e: u32, wk: u32x4) -> u32x4 {
+    vsha1cq_u32_(hash_abcd, hash_e, wk)
+}
+
+/// SHA1 fixed rotate.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha1h))]
+pub unsafe fn vsha1h_u32(hash_e: u32) -> u32 {
+    vsha1h_u32_(hash_e)
+}
+
+/// SHA1 hash update accelerator, majority.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha1m))]
+pub unsafe fn vsha1mq_u32(hash_abcd: u32x4, hash_e: u32, wk: u32x4) -> u32x4 {
+    vsha1mq_u32_(hash_abcd, hash_e, wk)
+}
+
+/// SHA1 hash update accelerator, parity.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha1p))]
+pub unsafe fn vsha1pq_u32(hash_abcd: u32x4, hash_e: u32, wk: u32x4) -> u32x4 {
+    vsha1pq_u32_(hash_abcd, hash_e, wk)
+}
+
+/// SHA1 schedule update accelerator, first part.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha1su0))]
+pub unsafe fn vsha1su0q_u32(w0_3: u32x4, w4_7: u32x4, w8_11: u32x4) -> u32x4 {
```

Comment:
I think these should use the `uint32x4_t` types nowadays, right @gnzlbg?

Reply:
Makes sense.  Fixed.